provider_test.go

```diff
@@ -0,0 +1,25 @@
+package main
+
+import (
+	"testing"
+	"strings"
+)
+
+func TestParseAzureEnvironment(t *testing.T) {
+	envNamesArray := []string{"AZURECHINACLOUD", "AZUREGERMANCLOUD", "AZUREPUBLICCLOUD", "AZUREUSGOVERNMENTCLOUD"}

```

Comment:
Can you add a test case for `""`, which should equal to `azure.PublicCloud`

Reply:
Will fix this asap and also do some testing on the PR.

Reply:
Done, added the test case. It is moved to a new file